[CollectionLayoutAttributes] Fix comparison

gambard opened the pull request:
Fix the comparison operator of MDCCollectionViewLayoutAttributes.
The MDCCollectionViewLayoutAttributes isEqual: method used a isEqual: on
objects that can be nil. This returns nil even if the two objects are nil.

closes #1264

Commit:
[CollectionLayoutAttributes] Fix comparison

Fix the comparison operator of MDCCollectionViewLayoutAttributes.
The MDCCollectionViewLayoutAttributes isEqual: method used a isEqual: on
objects that can be nil. This returns nil even if the two objects are nil.

closes #1264

Commit:
[CollectionLayoutAttributes] Fix comparison

Fix the comparison operator of MDCCollectionViewLayoutAttributes.
The MDCCollectionViewLayoutAttributes isEqual: method used a isEqual: on
objects that can be nil. This returns nil even if the two objects are nil.

closes #1264

Arcank committed:
[CollectionCells] Add accessibility hint in edit mode (#1258)

* [CollectionCells] Add accessibility hint in edit mode

When editing, selectable/deselectable cells now give an accessibility
hint: "Double tap to select.". This helps users to know what they can do
while in edit mode.

Related: https://github.com/material-components/material-components-ios/issues/1257

* Add tests

randallli commented on components/CollectionLayoutAttributes/src/MDCCollectionViewLayoutAttributes.m:
```diff
@@ -77,4 +79,8 @@ - (NSUInteger)hash {
          (NSUInteger)self.separatorLineHeight ^ (NSUInteger)self.shouldHideSeparators;
 }
 
++ (BOOL)is:(id)object1 equalTo:(id)object2 {
```
Lets add a comment explaining that your trying to capture the nil==nil case.

randallli commented on components/CollectionLayoutAttributes/src/MDCCollectionViewLayoutAttributes.m:
```diff
@@ -54,11 +54,13 @@ - (BOOL)isEqual:(id)object {
       (otherAttrs.shouldShowGridBackground != self.shouldShowGridBackground) ||
       (otherAttrs.sectionOrdinalPosition != self.sectionOrdinalPosition) ||
-      ![otherAttrs.backgroundImage isEqual:self.backgroundImage] ||
+      ![MDCCollectionViewLayoutAttributes is:otherAttrs.backgroundImage
```
This looks like a perfect case to expand (create) a unit test.

gambard committed:
[Collection] Fix text in header/footer example (#1266)

The header and the footer of the catalog example now display the correct text.

Commit:
[CollectionLayoutAttributes] Add comment and test

Add a comment for the is:equalTo: method and a test for equality with nil
objects.

Commit:
Merge branch 'background' of github.com:gambard/material-components-ios into background

gambard commented:
The fast forward did not go well. I am closing this pull request and opening a new one.
